### PR TITLE
Feat/error handling when media payload is rejected by nginx

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR attempts to handle the case where payloads are rejected by nginx. Currently, if a payload is too large, the request is immediately rejected by nginx, preventing our backend error handling from catching and throwing the correct 413 error. To account for this, we check for an empty response, which would indicate that our request did not make it to our backend server. I have simulated this locally by moving our cors initialization after the express limit has been set on the backend to force a network error for large files, since it's not possible to test this properly on local dev, but we should still ensure this is working on staging after merging.

This PR also adds some miscellaneous fixes. The list of these are:
- Change the placeholder text from categories to collections for pages in the page moving modal
- Set the placeholders correctly for the selects in ComponentSettingsModal so they have placeholder styling
- Fixed the error message of category from appearing on the same line as title in the ComponentSettingsModal
- Update the error message for large file size and the field titles for ComponentSettingsModal, to resolve https://github.com/isomerpages/isomercms-frontend/issues/295. Also partially addresses https://github.com/isomerpages/isomercms-frontend/issues/301

### CODE REVIEW

File: src/components/ComponentSettingsModal.jsx [hunk 1]
```diff
@@ -416,14 +416,23 @@ const ComponentSettingsModal = ({
...
                           label: generateInitialCategoryLabel(originalCategory, isCategoryDisabled),
-                        }}
+                        }
+                        : null
+                      }
                       options={allCategories}
                     />
                   </div>
-                  <span className={elementStyles.error}>{errors.category}</span>
+                  { errors.category &&
```
Comment (kwajiehao) [inline src/components/ComponentSettingsModal.jsx:430]: We need to do this for the permalink field as well 
Screenshot 2020-12-21 at 2 44 27 PM

**Review by kwajiehao — CHANGES_REQUESTED** [review]
Left a comment on error message rendering bug

File: src/components/media/MediaSettingsModal.jsx [hunk 2]
```diff
@@ -94,10 +94,10 @@ export default class MediaSettingsModal extends Component {
           <Toast notificationType='error' text={`Another ${type === 'image' ? 'image' : 'file'} with the same name exists. Please choose a different name.`}/>, 
           {className: `${elementStyles.toastError} ${elementStyles.toastLong}`}
         );
-      } else if (err?.response?.status === 413) {
-        // Error due to file size too large
+      } else if (err?.response?.status === 413 || err?.response === undefined) {
+        // Error due to file size too large - we receive 413 if nginx accepts the payload but it is blocked by our express settings, and undefined if it is blocked by nginx
         toast(
-          <Toast notificationType='error' text={`The ${type === 'image' ? 'image' : 'file'} is too large. Please choose a smaller ${type === 'image' ? 'image' : 'file'}.`}/>, 
+          <Toast notificationType='error' text={`Unable to upload as ${type === 'image' ? 'image' : 'file'} size is too large. Please reduce your ${type === 'image' ? 'image' : 'file'} size and try again.`}/>,
```
Comment (kwajiehao) [inline src/components/media/MediaSettingsModal.jsx:100]: "as the file size exceeds 5mb" instead of "size is too large"

Comment (alexanderleegs) [reply src/components/media/MediaSettingsModal.jsx]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/299/commits/684120b4c5838f026d524047aa49a05d80aa5645

File: src/components/ComponentSettingsModal.jsx (hunk 1 above)
Comment (alexanderleegs) [reply src/components/ComponentSettingsModal.jsx:430]: Good catch, resolved in https://github.com/isomerpages/isomercms-frontend/pull/299/commits/b215b7e3dbfcf6738a61693d140e42d941832afc

**Review by kwajiehao — APPROVED** [review]
lgtm

**Pull request merged**